load/loader.go

```diff
@@ -138,7 +142,41 @@ func (l *CommonBenchmarkRunner) postRun(wg *sync.WaitGroup, start *time.Time) {
 	// Wait for all workers to finish
 	wg.Wait()
 	end := time.Now()
-	l.summary(end.Sub(*start))
+	took := end.Sub(*start)
+	l.summary(took)
+	if len(l.BenchmarkRunnerConfig.ResultsFile) > 0 {
+		l.saveTestResult(took, *start, end)
+	}
+}
+
+func (l *CommonBenchmarkRunner) saveTestResult(took time.Duration, start time.Time, end time.Time) {
+	totals := make(map[string]interface{})
+	metricRate := float64(l.metricCnt) / took.Seconds()
+	totals["metricRate"] = metricRate
+	if l.rowCnt > 0 {
+		rowRate := float64(l.rowCnt) / float64(took.Seconds())
+		totals["rowRate"] = rowRate
+	}
```

Comment:
This part of the function does result transformation which should not be a part of the function whose purpose is to save the test result.

I suggest extracting it outside of this function.

Reply:
addressed in latest commit